config/routes.rb

```diff
@@ -123,7 +123,12 @@
     resource :search,    only: :show
     resource :dashboard, only: :show, constraints: { format: /html|atom/ }
     resources :profiles, only: :show
-    resource :profile, only: [:edit, :update]
+    resource :profile, only: [:edit, :update] do
+      member do
+        get :delete
+        delete :destroy, as: :destroy
```

Comment:
@qrush I like that idea, same approach is (was) used for yank/unyank => `deletions#create/destroy` [routes](https://github.com/rubygems/rubygems.org/blob/d6a762018a10af25f7180ee1018bb643c318fd3a/config/routes.rb#L74).